docs/HACKING.md

```diff
@@ -291,3 +291,41 @@ the `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE` environment variable like so:
 ```
 OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=quay.io/user/origin-release:v{version number} bin/openshift-install create cluster --log-level=debug
 ```
+
+## Hacking on `machine-os-content`
+
+If you own part of the operating system (from kernel to kubelet) you
+are part of the `machine-os-content`.  More information in [OSUpgrades.md].
+You will want a workflow for testing changes to a cluster.
+
+By far the simplest in a lot of cases is going to be to use `oc debug node/`
+to start by testing your changes on e.g. a single worker node.
+Commands to use here are e.g. `rpm-ostree usroverlay` and/or `rpm-ostree override replace`.
+The first one gives you a writable overlayfs on `/usr` and you can go to town
+replacing binaries there (e.g. `/usr/bin/crio`) or whatever.  If you want to install
+a test kernel in particular you'll need to use `rpm-ostree override replace`.
+
+A much more advanced (but "correct") flow is to build a custom `machine-os-content`
+container.  For that, see https://github.com/coreos/coreos-assembler/pull/489
+(A future iteration of this document will better describe this part)
+But let's assume you have a custom container and have pushed to a registry, for
+this example `quay.io/example/machine-os-content:latest`.
+
+The even more "official" way to do this would be to use the above guide
+to build a custom release image which has that content as an override.
```

Comment:
"The best but most time consuming way to do this..."  Then "This more closely mimics a real deployment of your changes to the cluster"